discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -376,39 +398,57 @@ def parse_sol_tx_batch(
         # if the thread pool executor completes successfully without raising an exception
         # the data is successfully fetched so we can add it to the db session and dispatch
         # events to challenge bus
-        with db.scoped_session() as session:
-            for play in plays:
-                session.add(
-                    Play(
-                        user_id=play.get("user_id"),
-                        play_item_id=play.get("track_id"),
-                        created_at=play.get("created_at"),
-                        source=play.get("source"),
-                        slot=play.get("slot"),
-                        signature=play.get("tx_sig"),
-                    )
-                )
-                if play.get("tx_sig") == last_tx_in_batch:
-                    # Cache the latest play from this batch
-                    # This reflects the ordering from chain
-                    most_recent_db_play = {
-                        "signature": play.get("tx_sig"),
-                        "slot": play.get("slot"),
-                        "timestamp": int(play.get("created_at").timestamp()),
-                    }
-                    cache_latest_sol_play_db_tx(redis, most_recent_db_play)
-
-        track_play_ids = [play["track_id"] for play in plays]
-        if track_play_ids:
-            redis.sadd(TRACK_LISTEN_IDS, *track_play_ids)
-
-        for event in challenge_bus_events:
-            challenge_bus.dispatch(
-                ChallengeEvent.track_listen,
-                event.get("slot"),
-                event.get("user_id"),
-                {"created_at": event.get("created_at")},
-            )
+
+    # Cache the latest play from this batch
+    # This reflects the ordering from chain
+    for play in plays:
+        if play.get("signature") == last_tx_in_batch:
+            most_recent_db_play = {
+                "signature": play.get("signature"),
+                "slot": play.get("slot"),
+                "timestamp": int(play.get("created_at").timestamp()),
+            }
+            cache_latest_sol_play_db_tx(redis, most_recent_db_play)
+            break
+
+    db_save_start = time.time()
+    logger.info(
+        f"index_solana_plays.py | DB | Saving test to DB, fetched batch tx details in {db_save_start - batch_start_time}"
+    )
+
+    with db.scoped_session() as session:
+        logger.info(
+            f"index_solana_plays.py | DB | Acquired session in {time.time() - db_save_start}"
+        )
+        session_execute_start = time.time()
+        # Save in bulk
+        session.execute(Play.__table__.insert().values(plays))
```

Comment:
didn't you saw raw sql ended up being faster?

Reply:
it was marginally faster between this and raw but I was concerned about sql injection / other vectors re @dmanjunath

Reply:
If there's no UGC i don't think injection is a concern

Reply:
that was my thought too actually - injection only a concern when users input content, good to know